scheduler/scheduler.go

```diff
@@ -10,6 +10,7 @@ import (
 	executionsdk "github.com/mesg-foundation/engine/sdk/execution"
 	workflowsdk "github.com/mesg-foundation/engine/sdk/workflow"
 	"github.com/mesg-foundation/engine/workflow"
+	"github.com/sirupsen/logrus"
 )
 
 // Scheduler manages the executions based on the definition of the workflows
```

Comment:
This package got bigger with PR and still, there is not a single test :/

Reply:
this actually makes the scheduler smaller and yes I agree it needs testing, this is based on the sdk and don't have any mock for it so it's hard to reproduce behaviors. But yes I will add some tests on another PR

Reply:
> and don't have any mock for it so it's hard to reproduce behaviors.

Please don't use mocks. Rather then mocks try to organize the code so it could be unit tested.

>  But yes I will add some tests on another PR

This functionality is in this PR so I think the tests should also be here. It's not the first pr about graph without testing and PR about filter has already been started. I'm ok to add them in PR about filters, but not later.